### DISCUSSION TITLE
[blocks]: Pricing page component block

### DISCUSSION THREAD

**Discussion by justerenoir** [shadcn-ui/ui#3328] [Blocks Request]
### Description

It would be great to have a pricing page component block that also has options for yearly vs monthly pricing.

**Comment by birobirobiro** [comment]
Examples:

https://shipixen.com/shadcn-pricing-page

https://pricing-page-shadcn.vercel.app/

**Reply by PaperPrototype** [reply]
This is exactly what I was about to suggest

**Comment by ten-x-dev** [comment]
Done: https://github.com/shadcn-ui/ui/pull/3349

**Comment by krisnaw** [comment]
This is what I'm currently working on: a clone of the Vercel pricing page. Let me know if you guys are interested, and I'll create a pull request (PR).
@justerenoir @birobirobiro 

<img width="998" alt="Screenshot 2024-04-04 at 16 45 46" src="https://github.com/shadcn-ui/ui/assets/1654121/768f7e98-de30-4a8f-be55-aed2f4978a09">

**Reply by ten-x-dev** [reply]
Looks beautiful!

You might want to check out the changes in [our PR](https://github.com/shadcn-ui/ui/pull/3349) because there is some unintuitive configuration you'll have to set up for it to work 🙏 

Specifically, make sure to export:

```ts
export const description =
  "A pricing page with 3 tiers. You can also customize it with 4 tiers or less than 3 tiers - it always looks good. The monthly plan is highlighted with a 20% discount nudging users to pick the annual plan."

export const iframeHeight = "640px"

export const containerClassName = "w-full h-full p-4 lg:p-0"
```

From your page. Took us a while to figure out.

**Reply by krisnaw** [reply]
Thanks for the heads up! @ten-x-dev  I'll definitely take a look at your PR for reference. Appreciate the guidance!

**Reply by PaperPrototype** [reply]
Only thing I would suggest is, change the buttons for each card to be
Free -> "Login / Signup"
Pro -> "Subscribe"
Enterprise -> "Contact Us"
